### PR TITLE
Backup controller with filter

### PR DESCRIPTION
**Pull request by PravinRanjan10**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:
1. List of resources tested: [for one/all namespaces]
* Deployments :(configs, service accounts)
* All configs
* All PVC
* All SC
* All Services
* All secrets
* All endpoints
* All Replicasets
* All statefulsets

### CODE REVIEW

File: controllers/backup/backupcontroller.go [hunk 1]
```diff
@@ -131,5 +130,269 @@ func (c *Controller) runBackup(key string) error {
...
+			"backup": utils.NamespaceAndName(backup),
+			"phase":  backup.Status.Phase,
+		}).Infof("Backup: %s is not New, so will not be processed", backup.Name)
+		return
+	}
+	c.controller.Enqueue(obj)
+}
+
+func (c *Controller) handleDel(obj interface{}) {
+	c.controller.Enqueue(obj)
```
Comment (sushanthakumar) [inline controllers/backup/backupcontroller.go:397]: invoke meta service delete functionality

File: apis/kahu/v1beta1/backup_location.go [hunk 2]
```diff
@@ -21,8 +23,9 @@ import (
 // BackupLocationSpec defines the desired state of BackupLocation
 type BackupLocationSpec struct {
 	// ProviderName is a 3rd party driver which inernally connect to respective storage
-	// +optional
-	ProviderName string `json:"providerName,omitempty"`
+	// +kubebuilder:validation:Required
```
Comment (sushanthakumar) [inline apis/kahu/v1beta1/backup_location.go:26]: only +required in enough

File: controllers/backup/backupcontroller.go [hunk 3]
```diff
@@ -17,14 +17,28 @@ limitations under the License.
...
 	"k8s.io/client-go/kubernetes"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
 	"github.com/soda-cdm/kahu/apis/kahu/v1beta1"
+	kahuv1beta1 "github.com/soda-cdm/kahu/apis/kahu/v1beta1"
```
Comment (sushanthakumar) [inline controllers/backup/backupcontroller.go:41]: Duplicate import, remove one of them

Comment (PravinRanjan10) [reply controllers/backup/backupcontroller.go]: done

File: utils/validations.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+var SupportedResourceList = []string{"deployments", "replicasets", "configmaps", "secrets", "servicesr",
```
Comment (sushanthakumar) [inline utils/validations.go:28]: servicesr should be services

File: utils/validations.go [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	}
+
+	return errs
+}
+
+func validateNamespaceName(ns string) []error {
+	var errs []error
+
+	if ns == "" {
+		return nil
```
Comment (AmitRoushan) [inline utils/validations.go:79]: IMO, Namespace name cant be empty

Comment (PravinRanjan10) [reply utils/validations.go]: done

File: utils/validations.go [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+	"endpoints", "storageclasses", "persistentvolumeclaims", "statefulsets",
+}
+
+func ValidateIncludesExcludes(includesList, excludesList []string) []error {
+	var errs []error
+
+	includeRes := sets.NewString(includesList...)
+	excludeRes := sets.NewString(excludesList...)
+
+	if includeRes.Len() == 1 && includeRes.Has("*") {
```
[the remark this answers is not in the document]
Comment (PravinRanjan10) [reply utils/validations.go]: done

File: utils/validations.go (hunk 4 above)
Comment (PravinRanjan10) [reply utils/validations.go]: done

**Review by sushanthakumar — APPROVED** [review]

**Review by AmitRoushan — APPROVED** [review]
LGTM

**Pull request merged**